File: bootstrap.ts
import "npm:zx/globals";
import { $ } from "npm:zx";
import { installPackage } from "../../../helpers.ts";

await installPackage({
  packageName: "tmux",
  packageManager: argv.pm,
});
const tpmDir = path.join(os.homedir(), ".tmux/plugins/tpm/");

if (fs.existsSync(tpmDir)) {
  console.log(chalk.blue("tpm already installed"));
} else {
  await spinner(
    chalk.green("installing tpm "),
    () => $`git clone https://github.com/tmux-plugins/tpm ${tpmDir}`,
  );
}

await spinner(
  chalk.green("installing tpm plugins"),
  () => $`${path.join(tpmDir, "bin/install_plugins")}`,
);

File: bootstrap.rb
#!/usr/bin/env ruby

require_relative File.expand_path '~/.dotfiles/helpers.rb'

package_manager = get_package_manager_arg
validate_package_manager package_manager

install_package package_manager, 'tmux'

tpm_dir = File.expand_path '~/.tmux/plugins/tpm'
if Dir.exist? tpm_dir
  puts 'tpm already installed'.noop
else
  puts 'cloning tpm'.doing
  stream_command "git clone https://github.com/tmux-plugins/tpm #{tpm_dir}"
end

puts 'installing tpm plugins'.doing
stream_command '~/.tmux/plugins/tpm/bin/install_plugins'
